pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -210,19 +217,16 @@ cpdef int64_t delta_to_nanoseconds(delta) except? -1:
         return get_timedelta64_value(ensure_td64ns(delta))
 
     if PyDelta_Check(delta):
-        try:
-            return (
-                delta.days * 24 * 3600 * 1_000_000
-                + delta.seconds * 1_000_000
-                + delta.microseconds
-            ) * 1000
-        except OverflowError as err:
-            raise OutOfBoundsTimedelta(*err.args) from err
+        microseconds = (
+            delta.days * 24 * 3600 * 1_000_000
+            + delta.seconds * 1_000_000
+            + delta.microseconds
+        )
+        return calc_int_int(operator.mul, microseconds, 1000)
```

Comment:
worth checking perf, but i expect that using operator.mul is slower than the status quo

Reply:
I think this can probably be reverted. This was a WIP effort to standardize the way `OverflowError`s were both raised and turned into `OutOfBoundsTimedelta`s.